### DISCUSSION TITLE
Fatal exception during connection.close()???

### DISCUSSION THREAD

**Discussion by MakoSDV** [psycopg/psycopg#689] [Q&A]
I'm trying to migrate some code that worked in a Python 3.7 environment to work properly with an up-to-date Python 3.11 environment.

For some reason that I haven't been able to figure out, whenever a `psycopg2` or `psycopg` `connection` is closed in the new environment, it causes Python to crash with a fatal exception code `0xc0000374`.

For example, this basic code will result in the crash:
```python
import psycopg2
URL = r"postgresql://user:pass@host/database"
conn = psycopg2.connect(URL)
conn.close()
```
```
Windows fatal exception: code 0xc0000374

Current thread 0x000012e4 (most recent call first):
  File "pMain.py", line 4 in <module>
```

Using `psycopg` instead gives slightly more info about the exception:
```python
import psycopg
URL = r"postgresql://user:pass@host/database"
conn = psycopg.connect(URL)
conn.close()
```
```
Windows fatal exception: code 0xc0000374

Current thread 0x00000c70 (most recent call first):
  File "C:\path\to\conda\env\Lib\site-packages\psycopg\connection.py", line 817 in close
  File "pMain.py", line 4 in <module>
```

Environment details:
Database: PostgreSQL 12.15

Python OS: Windows Server 2016

Updated Conda environment:
- python 3.11.6 h2628c8c_0_cpython conda-forge
- psycopg2 2.9.7 py311h2abc067_1 conda-forge
- psycopg 3.1.13  py311h9e3867f_0 conda-forge
- plus a bunch of other modules

The exception doesn't occur in the old Conda environment:
- python 3.7.16 h6244533_0
- psycopg2 2.9.3 py37hcd4344a_0
- plus a bunch of other modules

I'm guessing it's some incompatibility with this combination of things in the setup, but it's not obvious what the actual problem is.
I'm still digging into it, so I'll update this if I learn more.

I was hoping that maybe someone in the community has some ideas on how I can more intelligently investigate it besides setting up a bunch of different environment and version permutations.

Thanks.

**Comment by dvarrazzo** [comment]
The error happens on 

https://github.com/psycopg/psycopg/blob/93e915ada9c7b8a077afa47fe95e93a85e0e4465/psycopg/psycopg/connection.py#L817

Which is a point in which the libpq deallocats some memory. I am afraid you are in DLL hell. You might have more than one copy of the `libpq.dll` sitting around.

Conda is not supported either: I don't know what they do with a system. I think you should hear from them.
